client/src/app/employee/employee.component.ts

```diff
@@ -20,8 +20,8 @@ export class EmployeeComponent {
   uiState = '';
 
   constructor(
-    private employeeQuery: EmployeeQuery,
-    private employeeService: EmployeeService,
+    private employeeQuery: EmployeeDetailQuery,
```

Comment:
I've gotten in the habit of naming constructor dependencies as simply as possible, only increasing the specificity when there is ambiguity.

In this case, because this is the `EmployeeComponent`, I would name the params as follows:

```typescript
private detailQuery: EmployeeDetailQuery,
private detailService: EmployeeDetailService,
private listService: EmployeeListService,
private listQuery: EmployeeListQuery
```

I might even simplify it more and name the primary query and service for the component as `query` and `service`.

Let's say you had a `EmployeeDetailComponent` that needed the `Detail` injectables and the `List` service.

```typescript
private query: EmployeeDetailQuery,
private service: EmployeeDetailService,
private listService: EmployeeListService
```

I've found that by breaking up services and components into smaller units, the shorter names are more readable and differentiating them isn't hard, even though they don't have adjectives/nouns in the param names, because the place where they are being injected provides some context itself.

Of course, this only (potentially) works if the classes are kept small and focused.

Thoughts?

Reply:
I agree with this convention. It makes sense to remove employee from all of the service/query objects since in the context of these components we are always working with employees. I think the only time we would need the entity in the name is if for example we have a `PayrollDetailService` being injected into an `EmployeeDetailComponent`. I believe in the context of `EmployeeDetailComponent` removing detail from the name also makes sense since we are within the detail context. Adding an `EmployeeListService` as `listService` to the `EmployeeDetailcomponent` would then serve as an example to add more clarity to the name of the service since we would be outside of the scope of `EmployeeListComponent`. We can try to work this convention moving forward and see if any issues come up.

Reply:
Another way to get around the naming issues is to _not_ inject all the services/queries directly into your component.

Instead use the services/queries directly related to your component as the only external dependencies and inject the other services/queries into those.

Example:

```typescript
// employee-detail.component.ts

employee = this.query.employee;
companies = this.query.companies;

constructor(
  private readonly query: EmployeeDetailQuery,
  private readonly service: EmployeeDetailService) { }

onEmployeeDelete(employee: EmployeeDetail) {
  this.service.delete(employee);
}

onCompanyChange(company: Company) {
  this.service.updateCompany(company);
}
```

```typescript
// employee-detail.query.ts
readonly employee = this.select(s => s.employee);
readonly companies = this.companyQuery.selectAll();

constructor(
  private readonly companyQuery: CompanyQuery, 
  private readonly store: EmployeeDetailStore) { super(store); }
```

```typescript
// employee-detail.service.ts
constructor(
  private readonly http: HttpClient,
  private readonly store: EmployeeDetailStore,
  private readonly companyService: CompanyService) { }

delete(employee: EmployeeDetail) {
  this.http.delete('...');
}

updateCompany(company: Company) {
  const { employee } = this.store.getValue();

  this.http.update('...', employee)
    .tap(emp => this.store.update({ employee: emp }))
    .tap(() => this.companyService.refresh());
}
```

Reply:
I decided to do something similar to `updateCompany` in `EmployeeDetailService` by calling the `EmployeeListService` to refresh the list data when employee details are updated.

``` 
 update(employeeUpdate: EmployeeUpdate): void {
    this.store.setLoading(true);

    this.http
      .put<EmployeeDetail>(`${this.apiRootUrl}/Employees`, employeeUpdate)
      .subscribe({
        error: () =>
          this.toastr.error(`Could not update employee ${employeeUpdate.id}`),
        next: (detail) => {
          this.store.update(detail);
          this.listService.getEmployees();
          this.toastr.show('Employee sucessfully updated!');
        },
        complete: () => this.store.setLoading(false),
      });
  }
```

Reply:
I saw that - I think that's a nice approach.